pyfr/backends/base/types.py

```diff
@@ -102,11 +104,14 @@ def _pack(self, ary):
 
     def _unpack(self, ary):
         # Unpack from [blocked] AoSoA to SoA
-        ary = ary.swapaxes(0, 1)
-
-        if len(self.ioshape) > 2:
-            ary = ary.reshape(self.datashape)
-            ary = ary.swapaxes(-2, -3)
+        ary = ary.reshape(self.datashape)
+
+        if len(self.ioshape) == 2:
+            ary = ary.swapaxes(0, 1)
+        elif len(self.ioshape) == 3:
+            ary = ary.transpose((1, 3, 0, 2, 4))
```

Comment:
Might be possible to merge the two blocks through a pair of .swapaxes calls with negative arguments.

Reply:
Something along the lines of: `ary = np.moveaxis(ary.swapaxes(-2, -3), 0, -3)`